lib/simp/cli/config/items/action/set_grub_password_action.rb

```diff
@@ -37,7 +37,14 @@ def apply_summary
     end
 
     def set_password_grub(grub_hash)
-      result = execute("sed -i 's/password_pbkdf2 root.*$/password_pbkdf2 root #{grub_hash}/' /etc/grub.d/01_users")
+      grub_users = '/etc/grub.d/01_users'
+
+      if File.exist?(grub_users)
+        result = execute("sed -i 's/password_pbkdf2 root.*$/password_pbkdf2 root #{grub_hash}/' /etc/grub.d/01_users")
+      else
+        result = execute("echo 'password_pbkdf2 root #{grub_hash}' > /etc/grub.d/01_users")
```

Comment:
This is incorrect. The file needs to be executable and the command will not work because the correct PATH and/or other variables are not exported.  The contents of this file on my test system is as follows:
#!/bin/sh -e
cat << EOF
if [ -f \${prefix}/user.cfg ]; then
  source \${prefix}/user.cfg
  if [ -n "\${GRUB2_PASSWORD}" ]; then
    set superusers="root"
    export superusers
    password_pbkdf2 root \${GRUB2_PASSWORD}
  fi
fi
EOF